### PR TITLE
implement a uniqueness validation on page by resource

### PR DESCRIPTION
**Pull request by andrewroth**

### CODE REVIEW

File: app/models/page.rb [hunk 1]
```diff
@@ -7,5 +7,6 @@ class Page < AbstractPage
   validates :filename, presence: true
   validates :resource, presence: true
   validates :position, presence: true, uniqueness: {scope: :resource}
+  validates :filename, presence: true, uniqueness: {scope: :resource}
```
Comment (knutsenm) [inline app/models/page.rb:10]: Should delete line 7

File: spec/models/page_spec.rb [hunk 2]
```diff
@@ -37,6 +37,12 @@
     expect(result.errors["position"]).to include("has already been taken")
   end
 
+  it "cannot duplicate Resource ID and Filename" do
+    result = described_class.create(filename: "04_ThirdPoint.xml", resource_id: 1, structure: structure, position: 1)
+
+    expect(result.errors["filename"]).to include("has already been taken")
```
Comment (knutsenm) [inline spec/models/page_spec.rb:43]: OK, but this also breaks the `position` validator, doesn't it?

**Review by knutsenm — CHANGES_REQUESTED** [review]

File: app/models/page.rb (hunk 1 above)
Comment (andrewroth) [reply app/models/page.rb:7]: Thanks, I just pushed a fix. I also reordered the validations so the two scope ones are together, I think that looks a bit nicer

File: spec/models/page_spec.rb (hunk 2 above)
Comment (andrewroth) [reply spec/models/page_spec.rb:44]: True, doesn't affect the test, but I fixed it anyhow to make sure we're targeting only filename

**Review by knutsenm — APPROVED** [review]

**Review by frett — APPROVED** [review]
Tried it out on stage and everything is working correctly

**Comment by frett** [comment]
@andrewroth I'm going to merge this in because it looks good to go

**Pull request merged**